### ISSUE TITLE
[Keras] Support remaining merge modes/layers

### ISSUE THREAD

**Issue by maxpumperla** [deeplearning4j/deeplearning4j#5589]
Dot, as found here https://keras.io/layers/merge/#dot, and for legacy purposes "cos" merge mode in `Merge` layer.

**Comment by livsky87** [comment]
Hello @maxpumperla , Can you see the progress on this issue?

**Comment by maxpumperla** [comment]
sorry @livsky87 I'm not working on this project anymore.

**Comment by agibsonccc** [comment]
@livsky87  feel free to open up a new issue if you'd like.  If you can be more specific, (eg: go ahead and give me a model) I'd be happy to take a look. The next release should be going out soonish (End of month at latest) snapshots are already up.

**Comment by livsky87** [comment]
@maxpumperla , Thank you for reply :)

@agibsonccc, Thank you for answer.
I'd trying to apply deeplearning4j to our project.

First Issue, I'm trying to load keras model which class_name is `Functional`.
```
Exception in thread "main" org.deeplearning4j.nn.modelimport.keras.exceptions.InvalidKerasConfigurationException: Expected model class name Model (found Functional). For more information, see https://deeplearning4j.konduit.ai/keras-import/overview
```
Of course, I already refer to below link
* https://deeplearning4j.konduit.ai/keras-import/model-functional#getting-started-with-importing-keras-functional-models
* https://community.konduit.ai/t/cannot-load-keras-functional-model/687

I still can't find exact solution of it.

Second is... our model has `Dot merge layer`
Unfortunately I have found documentation that this is not supported. from https://deeplearning4j.konduit.ai/keras-import/supported-features#merge-layers

I wonder support plans for this feature?

If you think positively, we will register as a new issue.

Thanks!

**Comment by agibsonccc** [comment]
@livsky87  yes please do and feel free to link to this one. Dot Merge is definitely not something we have (mainly because it hasn't come up that much) but I can see about getting that in to the release.  If not that, our snapshots.

**Comment by livsky87** [comment]
@agibsonccc , great!

I will test it with the latest snapshot first.
Thank you

**Comment by agibsonccc** [comment]
@livsky87  again, MergeMode Cos does NOT exist on current snapshots. I said that right above. I proposed that would be the only alternative right now is me looking at this and implementing it.

**Comment by livsky87** [comment]
@agibsonccc , thanks for the clarify. As you mentioned, I made new issue for supporting dot merge layer (#9234)
